lib/Myriad.pm

```diff
@@ -670,6 +672,38 @@ method setup_tracing () {
     return;
 }
 
+=head2 setup_metrics
+
+Prepare L<Metrics::Any::Adapter> to collect metrics.
+
+=cut
+
+method setup_metrics () {
+    my $adapter = $config->metrics_adapter;
+    my $host = $config->metrics_host;
+    my $port = $config->metrics_port;
+
+    my $code = sub {
+        undef $metrics_adapter;
+        *Metrics::Any::Adapter::adapter = sub {
```

Comment:
should be outside `$code`, we only need to define this once - undefining `$metrics_adapter` is sufficient... however, better to move that inline, we shouldn't be closing over an instance slot for this.